core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Backend.java

```diff
@@ -81,10 +86,23 @@ public boolean print(byte[] content, JsonNode configuration) {
       }
       return false;
     }
+  },
+  DEBUG() {
+    @Override
+    public boolean print(byte[] content, JsonNode configuration) {
+      StringBuilder buffer = new StringBuilder();
+      buffer.append("Printing debug output:");
+      for (byte b : content) {
+        buffer.append(String.format(" %02x", b));
+      }
+      log.error(buffer.toString());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Invoke method(s) only conditionally. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

Reply:
Probably should be debug-level instead

Reply:
I had it that way in the beginning, but the default logging configuration makes it go nowhere. If someone sets up a debugging printer, I assume they really want to see the output.